src/robusta/core/discovery/top_service_resolver.py

```diff
@@ -67,6 +67,19 @@ def guess_cached_resource(cls, name: str, namespace: str) -> Optional[TopLevelRe
                 return cached_resource
         return None
 
+    @classmethod
+    def guess_workload_from_labels(cls, labels: Dict[Any, Any] = None) -> Optional[TopLevelResource]:
```

Comment:
I think this should be part of the `guess_cached_resource` method
I would just add this logic, in case the existing logic didn't work.
This way, it will also be available to all other places using the returned value

Reply:
on it